added ability for admin users to select whether they would like users…

RichardLanders opened the pull request:
@saeedmb8 @miltontony

Commit:
added ability for admin users to select whether they would like users to be able to enter email addresses or numbers into their username

maltem-za commented on molo/profiles/forms.py:
```diff
@@ -13,6 +15,30 @@
 
 
+REGEX_PHONE = r'.*?(\(?\d{3})? ?[\.-]? ?\d{3} ?[\.-]? ?\d{4}.*?'
+REGEX_EMAIL = r'([\w\.-]+@[\w\.-]+)'
```
Let's give people the option to override these in settings. Something like:

``` python
from django.conf import settings

REGEX_PHONE = settings.REGEX_PHONE if hasattr(settings, 'REGEX_PHONE') else r'.*?(\(?\d{3})? ?[\.-]? ?\d{3} ?[\.-]? ?\d{4}.*?'
```

See https://docs.djangoproject.com/en/1.9/topics/settings/#using-settings-in-python-code

maltem-za commented on molo/profiles/forms.py:
```diff
@@ -95,6 +147,37 @@ class Meta:
+            invalid_msg = 'phone number'
+
+        elif getattr(profile_settings, 'prevent_email_in_username', False):
+            invalid_msg = 'email address'
```
Everything up to here is a repetition of what's happening in `clean_username`. I would factor out the repeated code into, for example, `get_validation_msg_fragment`.

Then both `clean_alias` and `clean_username` can start with something like:

``` python
validation_msg_fragment = get_validation_msg_fragment()
```

which you can then use when you're formatting the text for your ValidationError.

maltem-za commented on molo/profiles/migrations/0004_add_profile_settings.py:
```diff
@@ -0,0 +1,25 @@
+        ),
+        migrations.AddField(
+            model_name='userprofilessettings',
+            name='prevent_number_in_username',
```
I would call this `prevent_phone_number_in_username` and change the `verbose_name` too so that it's clear what kind of number we're talking about here, ie: not just any number.

maltem-za commented on molo/profiles/models.py:
```diff
@@ -22,6 +22,12 @@ class UserProfilesSettings(BaseSetting):
+    prevent_number_in_username = models.BooleanField(
+        default=False,
+        editable=True,
+        verbose_name=_('Prevent number in username / display name'),
```
I would call this `prevent_phone_number_in_username` and change the `verbose_name` too so that it's clear what kind of number we're talking about here, ie: not just any number.

maltem-za commented on molo/profiles/models.py:
```diff
@@ -22,6 +22,12 @@ class UserProfilesSettings(BaseSetting):
         verbose_name=_('Mobile number required'),
     )
 
+    prevent_number_in_username = models.BooleanField(
```
I would call this `prevent_phone_number_in_username` and change the `verbose_name` too so that it's clear what kind of number we're talking about here, ie: not just any number. Remember to update the migration.

Commit:
renamed prevent_number_in_username to prevent_phone_number_in_username;
removed duplication of code;
added the ability to override the regex's in settings;

maltem-za commented:
:+1:
